chapters/03-package-setup.Rmd

```diff
@@ -20,6 +28,70 @@ collection of classic novels that are freely available from Project Gutenberg,
 we write a software package that counts and analyzes the word frequency
 distribution in any arbitrary body of text.
 
+## Outline of action steps
+
+* Why create an R package? 
+
+    * Text: Whenever you want to use data or functions over more than one project.
+    * Text: Advantages over sharing code and/or data files:
+        * Easy for other people to obtain and install
+        * Built-in documentation of functions and data
+        * Incorporate package-level meta data: authors, version, etc.
+
+* What is an R package?
+
+    * Text: Files that follow the R package conventions. usethis and devtools, 
+    designed to help you do this.  
+    * Exercise: Install a simple package from GitHub.
+    Give some code that: loads package, 
+    calls `xxx()` and data `yyy`.  
+    Examine structure of GitHub repo.   
+    Can you find where is the `xxx()` defined?  `yyy`? 
+    Where is the name of the package? 
+    
+* Creating a package    
+
+    * Text: Where should your package live? What should you call your package?
+    * Code: `usethis::create_package()`
+    * Text: `DESCRIPTION` is structured text that describes package, 
+    `R/` for R script files that define functions in the package. 
+    * Exercise: Repeat steps, e.g. `usethis::create_package("zipfs")`, 
+    edit `DESCRIPTION` to have you as author.
+ 
+
+* Package development workflow
+    
+    * Text: As an example, add a function to this package. 
+    Here, function is provided, next chapter, talks about how you
+    write functions. 
+    * Code: `use_r()`
+    * Text: How do we load the function and check it works?
+    * Code: `devtools::load_all()`
+    * Exercise: Practice workflow. Make small edit to function, reload, 
+    rerun function.
+    * Text: `load_all()` emulates installing and loading, but only
+    for current session.  Sometimes you want to actually install and load.
+    * Code: `devtools::install()`, `library()`
+    
+* Adding data to a package
+
+    * Text: Data live in `data/` as `.RData` files.  
+    * Code: `usethis::use_data("FIXME")` Here with simplified data. Restart, `devtools::load_all()`, `FIXME`
+    * Text: Should document how data was created. 
+    Put raw data and scripts that end in `usethis::use_data()` in `data-raw/`
+    * Code: `use_data_raw()` 
+    * Exercise: Provide script for getting dracula text from Gutenberg. 
```

Comment:
Is this for the reader to do? i.e. if they get stuck, is this a load-bearing step for the package development?

Reply:
I mean we provide the script, the learner has to put it in the right location, end it with `use_data("dracula")` and run it.

Reply:
Ok, sounds great. I don't think it's reasonable or fun to make the exercises only non-essential things, so I guess I was asking a more general question which you also answered!